tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,222 @@
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(func(g Gomega) {
+							g.Expect(kubectl.GetResourceCondition(controlPlaneNamespace, "istio", istioName)).Should(ContainElement(resourceReconcilied))
+						}).Should(Succeed(), "Istio CR should be reconciled")
+
+						Eventually(func(g Gomega) {
+							g.Expect(kubectl.GetResourceCondition(controlPlaneNamespace, "istio", istioName)).Should(ContainElement(resourceReady))
+						}).Should(Succeed(), "Istio CR should be Ready")
+
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						GinkgoWriter.Println("Istiod pod name: ", istiodPodName)
+
+						Eventually(kubectl.GetPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"), "Istiod should be Running")
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						var istio v1alpha1.Istio
+						output, _ := kubectl.GetResource(controlPlaneNamespace, "istio", istioName)
+
+						err := json.Unmarshal([]byte(output), &istio)
+						if err != nil {
+							Fail("Error unmarshalling the IstioCr")
+						}
+
+						Expect(istio.Spec.Version).To(Equal(version), "Istio CR version should match the applied version")
+					})
+
+					It("istio resource stopped reconciling", func() {
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(kubectl.GetPodLogs).WithArguments(controlPlaneNamespace, istiodPodName, "30s").ShouldNot(ContainSubstring("Reconciliation done"))
+						GinkgoWriter.Println("Istio Operator stopped reconciling")
+					})
+
+					It("only istiod is deployed", func() {
+						expectedDeployments := []string{"istiod"}
+
+						deploymentsList, err := kubectl.GetDeployments(controlPlaneNamespace)
+						if err != nil {
+							Fail("Error getting deployments")
+						}
+
+						Expect(deploymentsList).To(Equal(expectedDeployments), "Deployments List should contains only istiod")
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						daemonsetsList, err := kubectl.GetDaemonSets(namespace)
+						if err != nil {
+							Fail(fmt.Sprintf("Error getting daemonsets: %s", err))
+						}
+
+						if ocp == "true" {
+							Expect(daemonsetsList).To(ContainElement("istio-cni-node"), "DaemonSet List should contains the CNI DaemonSet")
+
+							podName, err := kubectl.GetPodFromLabel(namespace, "k8s-app=istio-cni-node")
+							if err != nil {
+								Fail("Error getting pod name from deployment")
+							}
+
+							Eventually(kubectl.GetPodPhase).WithArguments(namespace, podName).Should(Equal("Running"), "CNI DaemonSet should be Running")
+							GinkgoWriter.Println("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Expect(daemonsetsList).To(BeEmpty(), "DaemonSet List should empty because it's not OpenShift")
```

Comment:
```suggestion
							Expect(daemonsetsList).To(BeEmpty(), "CNI DaemonSet should not be deployed on OpenShift")
```

Make sure to write the descriptions from the perspective of the person running the test. Here, you say "DaemonSet List", but the person doesn't really know what this refers to (because the fact that the test gets the daemonset list is an implementation detail).